cli/azd/cmd/up.go

```diff
@@ -99,11 +97,6 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	}
 	defer func() { _ = infra.Cleanup() }()
 
-	err = u.provisioningManager.Initialize(ctx, u.projectConfig.Path, infra.Options)
```

Comment:
This means that `AZURE_SUBSCRIPTION_ID` and `AZURE_LOCATION` wouldn't be available early on in the `up` workflow and in `preprovision` hooks. It does also slightly degrade the UX if a template didn't explicitly required any parameters, that there's an interruption of prompting in the middle.

In our env variable [docs](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/manage-environment-variables#environment-variables-provided-by-azd), we say these variables are available "Right before an environment is provisioned for the first time", which would still be accurate.

Reply:
I do wonder if we want to revisit the operation of `package` before `provision` in #3392

Reply:
Based on feedback from @vhvb1989, this backcompat behavior is restored for now and will be decided separately in #3745